mantidproject/mantid
# CreateDetectorTable algorithm for Workbench

**Description of work.**
This PR adds a new algorithm called CreateDetectorTable which takes a workspace as input and outputs a table workspace containing detector information for the inputted workspace. This means that the ability to right-click a workspace and select "Show Detectors" is now possible in Workbench.

**To test:**
* In Workbench and MantidPlot, load a workspace, right-click it, and select "Show Detectors". Check that the table which displays in Workbench contains the same values as the one in MantidPlot.
* In the list of algorithms, there should now be one under Utility->Workspaces called CreateDetectorTable. Open the dialog and check the following things:
    * The WorkspaceIndices field accepts a comma-separated list of workspace indices and only the ones inputted are added to the output table workspace. If the field is left blank, all workspace indices from the workspace are used.
    * When the IncludeData checkbox is ticked, the output table workspace has two additional columns, "Data Value" and "Data Error" which show the first value from each spectrum.
    * The table workspace created by the algorithm has whatever name is inputted in the DetectorTableWorkspace field, or if no input is given, the name of the input workspace with "-Detectors" added.

Fixes #26867 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
